plugins/slack_webhooks/test/models/slack_webhook_notification_renderer_test.rb

```diff
@@ -96,4 +107,36 @@
     result = SlackWebhookNotificationRenderer.render(deploy, subject)
     result.wont_match /:.*:/
   end
+
+  context 'when there are no pull requests in the deploy' do
+    let(:pull_requests) { [] }
+
+    it 'renders a warning' do
+      deploy = stub("deploy",
+        short_reference: "xyz",
+        changeset: changeset,
+        pending?: true,
+        url: "http://sams.on/url")
+
+      result = SlackWebhookNotificationRenderer.render(deploy, subject)
+      result.must_include 'There are commits going live that did not go through a pull request'
+    end
+  end
+
+  context 'when the deploy is not pending or running' do
+    it 'does not display pull request information' do
+      deploy = stub("deploy",
+        short_reference: "xyz",
+        changeset: changeset,
+        pending?: false,
+        running?: false,
+        errored?: false,
+        failed?: false,
+        succeeded?: true,
+        url: "http://sams.on/url")
+
+      result = SlackWebhookNotificationRenderer.render(deploy, subject)
+      result.wont_match /Pull Requests/
```

Comment:
Ah easy, i just copied the matcher further up the file. Want me to put a PR with a change, or did you want to quickly push a commit with it?